### PR TITLE
Suite sparse (update for Linuxbrew)

### PR DESCRIPTION
**Pull request by davydden**
Provide extra CFLAGS.

### CODE REVIEW

**Comment by dpo** [comment]
@davydden Please hang on. The Linux builds on Travis are still not quite resolved and all these PRs are going to put considerable stress on the OSX test bot. We have to resolve the first few formulas before sending them all to BrewBot and Travis.

**Comment by dpo** [comment]
@davydden PS : many thanks for all your updates!

**Comment by davydden** [comment]
No problems, there is no hurry!

File: suite-sparse.rb [hunk 1]
```diff
@@ -40,6 +40,12 @@ def install
     make_args += ["METIS_PATH=",
                   "METIS=-L#{Formula["metis4"].opt_lib} -lmetis"] if build.with? "metis4"
 
+    # Add some flags for linux
+    # -DNTIMER is needed to avoid undefined reference to SuiteSparse_time
+    if not(OS.mac?)
+      make_args << "CF=-fPIC -O3 -fno-common -fexceptions -DNTIMER $(CFLAGS)"
+    end
+
```
Comment (dpo) [inline suite-sparse.rb]: Please make this a one liner.

File: suite-sparse.rb [hunk 2]
```diff
@@ -17,6 +17,7 @@ class SuiteSparse < Formula
 
   depends_on "tbb" => :recommended
   depends_on "openblas" => :optional
+  depends_on "veclibfort" => :optional if OS.mac?
```
Comment (dpo) [inline suite-sparse.rb]: `depends_on "veclibfort" if build.without?("openblas") && OS.mac?` (see, e.g., `arpack`).

Comment (davydden) [inline suite-sparse.rb]: it is not a mandatory dependence, so i guess at least something like `depends_on "veclibfort" => :optional if build.without?("openblas") && OS.mac?`

Comment (dpo) [inline suite-sparse.rb]: I think we should use `veclibfort` instead of `Accelerate` as it fixes some bugs.

**Comment by iml** [comment]
Please remember to squash commits when they're ready to be merged and prefix the message with the formula name. This applies to the other PRs as well.

**Comment by dpo** [comment]
Let's hope TravisCI comes back to life soon.

File: suite-sparse.rb (hunk 2 above)
Comment (davydden) [inline suite-sparse.rb]: I removed `veclibfort` as this is not  a fortran-only code, and `veclibfort` (https://github.com/mcg1969/vecLibFort) is aimed to fix issues when compiling `fortran` code, as far as I understand

**Comment by dpo** [comment]
@BrewTestBot test this please

**Comment by dpo** [comment]
AMD is Fortran and it will be compiled with `gfortran`.

**Comment by davydden** [comment]
yes, but there is C code as well (most of it).

**Comment by dpo** [comment]
The entire library doesn't have to be Fortran to make `veclibfort` relevant. Fortunately, `amd.f` doesn't appear to make BLAS calls directly, so I think we're ok.

**Comment by davydden** [comment]
i was not sure if `veclibfort` can be provided to link `C`/`C++` files.

**Comment by dpo** [comment]
`veclibfort` is a thin interface to `Accelerate` that shadows just a few BLAS functions.

**Pull request closed without merging**